src/nimble.nim

```diff
@@ -851,14 +851,20 @@ proc uninstall(options: Options) =
       raise newException(NimbleError, "Package not found")
 
     display("Checking", "reverse dependencies", priority = HighPriority)
-    var errors: seq[string] = @[]
     for pkg in pkgList:
       # Check whether any packages depend on the ones the user is trying to
       # uninstall.
       if options.uninstallRevDeps:
         getAllRevDeps(options, pkg, pkgsToDelete)
       else:
-        let revDeps = getRevDeps(options, pkg)
+        let
+          # Pkgs already processed for removal
+          namesOfPkgsToDelete = pkgsToDelete.mapIt(it.name)
```

Comment:
This is a good start, but it doesn't take package versions into account at all. Users can specify version requirements when removing packages: ``nimble remove foo@0.1.0``.

I think what might be a good approach to this is to refactor ``pkgsToDelete`` into a ``HashSet[PackageInfo]``, you'll need to implement a ``hash()`` function for ``PackageInfo`` and ensure the hash is based on the package name + version. If all goes well you'll be able to write ``if len(revDeps * pkgsToDelete) > 0:`` below instead of ``revDeps.len > 0`` and this mapping/filtering logic.

Reply:
I've updated per our discussion in IRC. Had to be `revDeps - pkgsToDelete`.